pages/[category]/index.tsx

```diff
@@ -1,51 +1,236 @@
+  const [citySelected, setCitySelected] = useState<string[]>([])
+  const [serviceSelected, setServiceSelected] = useState<string[]>([])
+  const [genderSelected, setGenderSelected] = useState<string[]>([])
+  const [languageSelected, setLanguageSelected] = useState<string[]>([])
+  const [filteredResults, setFilteredResults] = useState<any | null>([])
+  const [open, setOpen] = useState(false)
+  const [currentView, setCurrentView] = useState('list')
+  const [modalStyle] = useState(getModalStyle)
+  const activeCopy = categoryCopy[language]
+  const setFunctions = {
+    setServiceSelected,
+    setGenderSelected,
+    setLanguageSelected,
+    setCitySelected,
+  }
   const validCategory = categories[asPath]
-
-  if (!validCategory) return <Error statusCode={404} />
-
-  const displayCategory: string = validCategory[language].category
-  const routeCategory: string = validCategory.english.category.toLowerCase()
-
+  const routeCategory: string = validCategory?.english.category.toLowerCase()
+  const displayCategory: string = validCategory?.[language].category
+  const displayDescription: string = validCategory?.[language].description
   const { fetchedRecords, setFetchedRecords } =
     useMultipleListRecords(routeCategory)
   const { convertedLocRecords, setLocationRecords } =
     useConvertedLocationRecords()
-
   useEffect((): void => {
-    if (fetchedRecords) setLocationRecords(fetchedRecords)
-  }, [fetchedRecords])
-
+    let keywordQuery = serviceSelected.concat(
+      citySelected,
+      genderSelected,
+      languageSelected,
+    )
+    if (fetchedRecords && keywordQuery.length > 0) {
+      let newResults = getMatchingRecords(
+        fetchedRecords,
+        serviceSelected.concat(
+          citySelected,
+          serviceSelected,
+          genderSelected,
+          languageSelected,
+        ),
+      )
+      setFilteredResults(newResults)
+      setLocationRecords(newResults)
+    } else if (fetchedRecords && keywordQuery.length === 0) {
+      setFilteredResults(fetchedRecords)
+      setLocationRecords(fetchedRecords)
+    }
+  }, [
+    fetchedRecords,
+    serviceSelected,
+    citySelected,
+    genderSelected,
+    languageSelected,
+    validCategory,
+  ])
+  if (!validCategory) return <Error statusCode={404} />
+  const handleSelected = e => {
+    let value = e.target.value
+    setFunctions[e.target.name](value)
+  }
   return (
     <>
       <Head>
         <title>{`${siteTitle} | ${displayCategory}`}</title>
       </Head>
-      <RecordPane
-        orgInfo={fetchedRecords}
-        displayCategory={displayCategory}
-        routeCategory={routeCategory}
-        setRecords={setFetchedRecords}
-      />
-
-      {convertedLocRecords && <DisplayMap latLongInfo={convertedLocRecords} />}
+      <div className={classes.root}>
+        <Grid container spacing={3}>
+          <Grid item xs={12}>
+            <Title>{displayCategory}</Title>
+          </Grid>
+          <Grid item xs={12} sm={12} md={4}>
```

Comment:
There are some odd and (I think) conflicting css behaviors that are making this behave a bid oddly https://www.loom.com/share/650c13ffffc94c6897f8282681d7329d